//tensorflow/compiler/xla/tools:interactive_graphviz_build_only_test fails to build

<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Source

source

### Tensorflow Version

git HEAD

### Custom Code

No

### OS Platform and Distribution

Ubuntu 20.04

### Mobile device

n/a

### Python version

3.8.10

### Bazel version

5.3.0

### GCC/Compiler version

10.2.1

### CUDA/cuDNN version

n/a

### GPU model and memory

n/a

### Current Behaviour?

```shell
Trying to build the test //tensorflow/compiler/xla/tools:interactive_graphviz_build_only_test fails with link errors such as
`bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::_AlgorithmProto_default_instance_'`
```


### Standalone code to reproduce the issue

```shell
bazel test --test_timeout=300,500,-1,-1 --flaky_test_attempts=2 --test_output=all --cache_test_results=no --config=nonccl --config=mkl_aarch64_threadpool --copt=-mtune=generic --copt=-march=armv8-a --copt=-O3 --test_env=TF_ENABLE_ONEDNN_OPTS=1 --verbose_failures --build_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-no_aarch64,-requires-gpu --test_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-no_aarch64,-requires-gpu --build_tests_only -- //tensorflow/compiler/xla/tools:interactive_graphviz_build_only_test
```


### Relevant log output

```shell
bazel test --test_timeout=300,500,-1,-1 --flaky_test_attempts=2 --test_output=all --cache_test_results=no --config=nonccl --config=mkl_aarch64_threadpool --copt=-mtune=generic --copt=-march=armv8-a --copt=-O3 --test_env=TF_ENABLE_ONEDNN_OPTS=1 --verbose_failures --build_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-no_aarch64,-requires-gpu --test_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-no_aarch64,-requires-gpu --build_tests_only -- //tensorflow/compiler/xla/tools:interactive_graphviz_build_only_test
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=117
INFO: Reading rc options for 'test' from /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc:
  Inherited 'common' options: --experimental_repo_remote_exec
INFO: Reading rc options for 'test' from /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc:
  Inherited 'build' options: --define framework_shared_object=true --define tsl_protobuf_header_only=true --define=use_fast_cpp_protos=true --define=allow_oversize_protos=true --spawn_strategy=standalone -c opt --announce_rc --define=grpc_no_ares=true --noincompatible_remove_legacy_whole_archive --enable_platform_specific_config --define=with_xla_support=true --config=short_logs --config=v2 --define=no_aws_support=true --define=no_hdfs_support=true --experimental_cc_shared_library --experimental_link_static_libraries_once=false
INFO: Reading rc options for 'test' from /home/builder/1/tensorflow_build/tensorflow-git/.tf_configure.bazelrc:
  Inherited 'build' options: --action_env PYTHON_BIN_PATH=/home/builder/1/tensorflow_build/venv_py38/bin/python3 --action_env PYTHON_LIB_PATH=/home/builder/1/tensorflow_build/venv_py38/lib/python3.8/site-packages --python_path=/home/builder/1/tensorflow_build/venv_py38/bin/python3
INFO: Reading rc options for 'test' from /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc:
  Inherited 'build' options: --deleted_packages=tensorflow/compiler/mlir/tfrt,tensorflow/compiler/mlir/tfrt/benchmarks,tensorflow/compiler/mlir/tfrt/jit/python_binding,tensorflow/compiler/mlir/tfrt/jit/transforms,tensorflow/compiler/mlir/tfrt/python_tests,tensorflow/compiler/mlir/tfrt/tests,tensorflow/compiler/mlir/tfrt/tests/ir,tensorflow/compiler/mlir/tfrt/tests/analysis,tensorflow/compiler/mlir/tfrt/tests/jit,tensorflow/compiler/mlir/tfrt/tests/lhlo_to_tfrt,tensorflow/compiler/mlir/tfrt/tests/lhlo_to_jitrt,tensorflow/compiler/mlir/tfrt/tests/tf_to_corert,tensorflow/compiler/mlir/tfrt/tests/tf_to_tfrt_data,tensorflow/compiler/mlir/tfrt/tests/saved_model,tensorflow/compiler/mlir/tfrt/transforms/lhlo_gpu_to_tfrt_gpu,tensorflow/core/runtime_fallback,tensorflow/core/runtime_fallback/conversion,tensorflow/core/runtime_fallback/kernel,tensorflow/core/runtime_fallback/opdefs,tensorflow/core/runtime_fallback/runtime,tensorflow/core/runtime_fallback/util,tensorflow/core/tfrt/common,tensorflow/core/tfrt/eager,tensorflow/core/tfrt/eager/backends/cpu,tensorflow/core/tfrt/eager/backends/gpu,tensorflow/core/tfrt/eager/core_runtime,tensorflow/core/tfrt/eager/cpp_tests/core_runtime,tensorflow/core/tfrt/gpu,tensorflow/core/tfrt/run_handler_thread_pool,tensorflow/core/tfrt/runtime,tensorflow/core/tfrt/saved_model,tensorflow/core/tfrt/graph_executor,tensorflow/core/tfrt/saved_model/tests,tensorflow/core/tfrt/tpu,tensorflow/core/tfrt/utils
INFO: Reading rc options for 'test' from /home/builder/1/tensorflow_build/tensorflow-git/.tf_configure.bazelrc:
  'test' options: --flaky_test_attempts=3 --test_size_filters=small,medium
INFO: Found applicable config definition build:short_logs in file /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc: --output_filter=DONT_MATCH_ANYTHING
INFO: Found applicable config definition build:v2 in file /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc: --define=tf_api_version=2 --action_env=TF2_BEHAVIOR=1
INFO: Found applicable config definition test:v2 in file /home/builder/1/tensorflow_build/tensorflow-git/.tf_configure.bazelrc: --test_tag_filters=-benchmark-test,-no_oss,-gpu,-oss_serial,-v1only --build_tag_filters=-benchmark-test,-no_oss,-gpu,-v1only
INFO: Found applicable config definition build:nonccl in file /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc: --define=no_nccl_support=true
INFO: Found applicable config definition build:mkl_aarch64_threadpool in file /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc: --define=build_with_mkl_aarch64=true -c opt
INFO: Found applicable config definition build:linux in file /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc: --host_copt=-w --copt=-Wno-all --copt=-Wno-extra --copt=-Wno-deprecated --copt=-Wno-deprecated-declarations --copt=-Wno-ignored-attributes --copt=-Wno-unknown-warning --copt=-Wno-array-parameter --copt=-Wno-stringop-overflow --copt=-Wno-array-bounds --copt=-Wunused-result --copt=-Werror=unused-result --copt=-Wswitch --copt=-Werror=switch --define=PREFIX=/usr --define=LIBDIR=$(PREFIX)/lib --define=INCLUDEDIR=$(PREFIX)/include --define=PROTOBUF_INCLUDE_PATH=$(PREFIX)/include --cxxopt=-std=c++17 --host_cxxopt=-std=c++17 --config=dynamic_kernels --distinct_host_configuration=false --experimental_guard_against_concurrent_changes
INFO: Found applicable config definition build:dynamic_kernels in file /home/builder/1/tensorflow_build/tensorflow-git/.bazelrc: --define=dynamic_loaded_kernels=true --copt=-DAUTOLOAD_DYNAMIC_KERNELS
INFO: Analyzed target //tensorflow/compiler/xla/tools:interactive_graphviz_build_only_test (1 packages loaded, 7 targets configured).
INFO: Found 1 test target...
ERROR: /home/builder/1/tensorflow_build/tensorflow-git/tensorflow/compiler/xla/tools/BUILD:255:14: Linking tensorflow/compiler/xla/tools/interactive_graphviz failed: (Exit 1): gcc failed: error executing command 
  (cd /home/builder/.cache/bazel/_bazel_builder/9dc2dbd69dc3512cedb530e1521082e7/execroot/org_tensorflow && \
  exec env - \
    PATH=/home/builder/.cache/bazelisk/downloads/bazelbuild/bazel-5.3.0-linux-arm64/bin:/home/builder/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin \
    PWD=/proc/self/cwd \
    PYTHON_BIN_PATH=/home/builder/1/tensorflow_build/venv_py38/bin/python3 \
    PYTHON_LIB_PATH=/home/builder/1/tensorflow_build/venv_py38/lib/python3.8/site-packages \
    TF2_BEHAVIOR=1 \
  /usr/local/bin/gcc @bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/tools/interactive_graphviz-2.params)
# Configuration: d45cf8a96c93cdeab0dbdaf20276fe89827c573a72a6bad81998b5c180ddbd8c
# Execution platform: @local_execution_config_platform//:platform
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::_AlgorithmProto_default_instance_'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::_AlgorithmProto_default_instance_'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::AlgorithmProto(stream_executor::dnn::AlgorithmProto const&)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::AlgorithmProto()'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::InternalSwap(stream_executor::dnn::AlgorithmProto*)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::~AlgorithmProto()'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::~AlgorithmProto()'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::~AlgorithmProto()'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/_objs/hlo_graph_dumper/hlo_graph_dumper.o:hlo_graph_dumper.cc:function xla::(anonymous namespace)::HloDotDumper::GetInstructionNodeBackendConfig(xla::HloInstruction const*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::~AlgorithmProto()'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function InitDefaultsscc_info_CudnnConvBackendConfig_tensorflow_2fcompiler_2fxla_2fservice_2fgpu_2fbackend_5fconfigs_2eproto(): error: undefined reference to 'stream_executor::dnn::_AlgorithmProto_default_instance_'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function InitDefaultsscc_info_CudnnConvBackendConfig_tensorflow_2fcompiler_2fxla_2fservice_2fgpu_2fbackend_5fconfigs_2eproto(): error: undefined reference to 'stream_executor::dnn::_AlgorithmProto_default_instance_'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::MergeFrom(xla::gpu::CudnnConvBackendConfig const&): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::MergeFrom(stream_executor::dnn::AlgorithmProto const&)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::MergeFrom(xla::gpu::CudnnConvBackendConfig const&): error: undefined reference to 'stream_executor::dnn::AlgorithmProto* google::protobuf::Arena::CreateMaybeMessage<stream_executor::dnn::AlgorithmProto>(google::protobuf::Arena*)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::MergeFrom(xla::gpu::CudnnConvBackendConfig const&): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::MergeFrom(stream_executor::dnn::AlgorithmProto const&)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::MergeFrom(google::protobuf::Message const&): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::MergeFrom(stream_executor::dnn::AlgorithmProto const&)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::MergeFrom(google::protobuf::Message const&): error: undefined reference to 'stream_executor::dnn::AlgorithmProto* google::protobuf::Arena::CreateMaybeMessage<stream_executor::dnn::AlgorithmProto>(google::protobuf::Arena*)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::MergeFrom(google::protobuf::Message const&): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::MergeFrom(stream_executor::dnn::AlgorithmProto const&)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::MergePartialFromCodedStream(google::protobuf::io::CodedInputStream*): error: undefined reference to 'stream_executor::dnn::AlgorithmProto* google::protobuf::Arena::CreateMaybeMessage<stream_executor::dnn::AlgorithmProto>(google::protobuf::Arena*)'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::InternalSerializeWithCachedSizesToArray(unsigned char*) const: error: undefined reference to 'stream_executor::dnn::AlgorithmProto::InternalSerializeWithCachedSizesToArray(unsigned char*) const'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:function xla::gpu::CudnnConvBackendConfig::ByteSizeLong() const: error: undefined reference to 'stream_executor::dnn::AlgorithmProto::ByteSizeLong() const'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/service/gpu/_objs/backend_configs_cc_impl/backend_configs.pb.o:backend_configs.pb.cc:scc_info_CudnnConvBackendConfig_tensorflow_2fcompiler_2fxla_2fservice_2fgpu_2fbackend_5fconfigs_2eproto: error: undefined reference to 'scc_info_AlgorithmProto_tensorflow_2ftsl_2fprotobuf_2fdnn_2eproto'
bazel-out/aarch64-opt/bin/tensorflow/compiler/xla/stream_executor/_objs/dnn_proto_cc_impl/dnn.pb.o:dnn.pb.cc:descriptor_table_tensorflow_2fcompiler_2fxla_2fstream_5fexecutor_2fdnn_2eproto_deps: error: undefined reference to 'descriptor_table_tensorflow_2ftsl_2fprotobuf_2fdnn_2eproto'
collect2: error: ld returned 1 exit status
Target //tensorflow/compiler/xla/tools:interactive_graphviz_build_only_test failed to build
INFO: Elapsed time: 33.177s, Critical Path: 32.38s
INFO: 3 processes: 2 internal, 1 local.
FAILED: Build did NOT complete successfully
//tensorflow/compiler/xla/tools:interactive_graphviz_build_only_test FAILED TO BUILD

FAILED: Build did NOT complete successfully
```
</details>

--- a/tensorflow/compiler/xla/tools/BUILD
+++ b/tensorflow/compiler/xla/tools/BUILD
@@ -273,6 +273,7 @@ xla_cc_binary(
         "//tensorflow/tsl/platform:path",
         "//tensorflow/tsl/platform:platform_port",
         "//tensorflow/tsl/platform:subprocess",
+        "//tensorflow/tsl/protobuf:dnn_proto_cc_impl",
         "//tensorflow/tsl/protobuf:error_codes_proto_impl_cc",
         "//tensorflow/tsl/util:command_line_flags",
     ] + if_cuda_or_rocm([

